core/trino-spi/src/main/java/io/trino/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -1739,6 +1739,8 @@ private static void copyUnscaledDecimal(Slice from, Slice to)
 
     private static void pack(Slice decimal, int v0, int v1, int v2, int v3, boolean negative)
     {
+        // check if the result is zero and if so, don't create negative zeros:
+        negative &= v0 != 0 || v1 != 0 || v2 != 0 || v3 != 0;
```

Comment:
`v0 != 0 || v1 != 0 || v2 != 0 || v3 != 0;` contains 4 branches while
`(v0 | v1 | v2 | v3) != 0` only two.

Reply:
I restored the `|`'s but without the comment